### DISCUSSION TITLE
Linkerd version 2 routing

### DISCUSSION THREAD

**Discussion by pisees** [linkerd/linkerd2#5787] [Q&A]
Does linkerd version 2 still support routing with dtab?

**Comment by cpretzer** [comment]
Hi @pisees,

Linkerd 2 doesn't have support for dtabs because the architecture is completely different. 

Specifically, Linkerd 2 doesn't use finagle and dtabs are a concept defined by finagle.

**Reply by pisees** [reply]
Thanks @cpretzer 
So to confirm linkerd2 does not support routing (for example header based).

Are there any recommendations how to support routing, north/south and east/west?

**Reply by cpretzer** [reply]
That's right, today Linkerd doesn't support header based routing, but there is an issue tracking that feature request #3165

For north/south traffic, I'd use an ingress controller. For east/west traffic, you could use a proxy (ambassador, nginx, etc.) to route traffic through, like a hub and spoke model.

If yoiu're interested in helping to build header based routing, that would be awesome! :smile:
